src/unittest/test_filesys.cpp

```diff
@@ -262,3 +265,14 @@ void TestFilePath::testRemoveRelativePathComponent()
 	result = fs::RemoveRelativePathComponents(path);
 	UASSERT(result == p("/a/e"));
 }
+
+
+void TestFileSys::testSafeWriteToFile()
+{
+	const std::string dest_path = fs::TempPath() + DIR_DELIM + "testSafeWriteToFile.txt";
+	fs::safeWriteToFile(dest_path, "test");
+	UASSERT(fs::PathExists(dest_path));
+	std::string contents_actual;
+	UASSERT(fs::ReadFile(dest_path, contents_actual));
+	UASSERT(contents_actual == "test");
```

Comment:
worth testing that the implementation doesn't accidentally use strlen
```suggestion
	const std::string dest_path = fs::TempPath() + DIR_DELIM + "testSafeWriteToFile.txt";
	constexpr std::string test_data("hello\0world", 11);
	fs::safeWriteToFile(dest_path, test_data);
	UASSERT(fs::PathExists(dest_path));
	std::string contents_actual;
	UASSERT(fs::ReadFile(dest_path, contents_actual));
	UASSERT(contents_actual == test_data);
```